x/blob/README.md

```diff
@@ -2,44 +2,141 @@
 
-Since `BlobTx`s can contain multiple blobs, the `BlobTx` is wrapped with one share index per blob in the transaction. The index wrapped transaction is called an [IndexWrapper](https://github.com/celestiaorg/celestia-core/blob/2d2a65f59eabf1993804168414b86d758f30c383/proto/tendermint/types/types.proto#L192-L198) and this is the type that gets marshalled and written to the PayForBlobNamespace.
+1. Split the blob into shares of size `appconsts.ShareSize`
+1. Determine the
+   [`SubtreeWidth`](https://github.com/celestiaorg/celestia-app/blob/v1.0.0-rc2/pkg/shares/non_interactive_defaults.go#L94-L116)
+   by dividing the length in shares by the `SubtreeRootThreshold`.
+1. Generate each subtree root by diving the blob shares into `SubtreeWidth`
+   sized sets, then take the binary [namespaced merkle tree
+   (NMT)](https://github.com/celestiaorg/nmt/blob/v0.16.0/docs/spec/nmt.md) root
+   of each set of shares.
+1. Calculate the final share commitment by taking the merkle root (note: not an
+   NMT, just a normal binary merkle root) of the subtree roots from the previous
+   step.
+
+See
+[`CreateCommitment`](https://github.com/celestiaorg/celestia-app/blob/v1.0.0-rc2/x/blob/types/payforblob.go#L169-L236)
+for an implementation. See [data square
+layout](../../specs/src/specs/data_square_layout.md) and
+[ADR013](../../docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md)
+for details on the rational of the square layout.
+
+## Validity Rules
+
+In order for a proposal block to be considered valid, each `BlobTx`, and thus
+each PFB, to be included in a block must follow a set of validity rules.
+
+1. Signatures: All blob transactions must have valid signatures. This is
+   state-dependent because correct signatures require using the correct sequence
+   number(aka nonce).
+1. Single SDK.Msg: There must be only a single sdk.Msg in the blob transaction.
+1. Namespace Validity: The namespace of each blob transaction must be valid.
+   This validity is determined by the following sub-rules:
+    1. The namespace is not within the reserved namespace range.
+    1. The namespace is not the tail padding or parity namespaces.
+1. Blob Size: No blob can have a size of 0.
+1. Blob Count: There must be one or more blobs included in the transaction.
+1. Share Commitment Validity: Each share commitment must be valid.
+    1. The size of each of the share commitments must be equal to the digest of
+       the hash function used (sha256 so 32 bytes).
+    1. The share commitment must be calculated using the steps specified above
+       in [Generating the Share
+       Commitment](./README.md#generating-the-sharecommitment)
+1. Share Versions: The versions of the shares must be supported.
+1. Signer Address: The signer address must be a valid Celestia address.
+1. Proper Encoding: The blob transactions must be properly encoded.
+1. Size Consistency: The sizes included in the transaction must match the actual
+   sizes of the blobs.
```

Comment:
A bit of elaboration on this would be helpful for accuracy, e.g., when we say, is it the byte size or the bit size? also, the exact name of the field of the `BlobTx` to which this rule applies.

Reply:
:+1: https://github.com/celestiaorg/celestia-app/pull/1906/commits/4338e36444725cb94055ff3c406181e29a541571